### PR TITLE
docker

### PR DESCRIPTION
**Pull request by nothub**
This PR contains some changes and additions to Docker related stuff:
- updated Dockerfile
- added docker-compose.yml
- switched .dockerignore to whitelisting
- added readme section

Because Dockers init system is not active by default, its nice to have something like tini in the image.

### CODE REVIEW

**Comment by hugmanrique** [comment]
I'm not familiar with tini, but its documentation specifies Docker includes a similar init system already https://github.com/krallin/tini#using-tini. Could we use that instead?

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+services:
+  minetrack:
+    build: .
+    container_name: minetrack
+    dns:
+      - 8.8.8.8
+      - 1.1.1.1
+    ports:
+      - "80:8080"
```
Comment (hugmanrique) [inline docker-compose.yml:11]: What's the reasoning about changing 8080 to 80 as the host default port?

**Comment by hugmanrique** [comment]
I'm torn about changing the .dockerignore. Everything else mostly looks good to me. Thanks for your contribution 😀

File: docker-compose.yml (hunk 1 above)
Comment (nothub) [reply docker-compose.yml:11]: I just did this change so its more convenient for users when they run the commands without thinking about it. Im totally neutral here. If you think it is better overall to use the original port as external port, i gonna change it.

**Comment by nothub** [comment]
> I'm not familiar with tini, but its documentation specifies Docker includes a similar init system already https://github.com/krallin/tini#using-tini. As you say this isn't active by default, but we could document using the --init flag is recommended. Could we use that instead?

There are certain runtimes (Kubernetes, etc.) that do not include an init system but are also capable of running docker containers. Since most users that run containers on Kubernetes will hopefully know about this, it should not be a big difference.

**Comment by nothub** [comment]
> I'm torn about changing the .dockerignore. Everything else mostly looks good to me. Thanks for your contribution grinning

The problem i see here is, that a container built without explicitly ignoring files from the build, will have an unnecessary bigger image size. Another way to achieve this would either be to use the .dockerignore file as blacklist (would need additional changes in the future if new files are introduced to the repo) or explicitly run a COPY for every needed filepath (every COPY introduces a new layer to the image).

**Comment by hugmanrique** [comment]
Yes, was talking about the whitelist-vs-blacklist approach here. You bring a good point about ignoring future non-source code files. Your approach is ideal for projects with a unique source code path, usually `src`. Minetrack doesn't (currently) follow that convention, and source code additions are more common than build ones. Is this an established pattern? I'm only used to `.gitignore`s, where the blacklist approach is the standard.

Re port: most people run Minetrack behind a Nginx proxy on the same machine. However, I doubt this is the case on a K8s setup. I'm okay-ish with it, as long as it's documented in the readme.

Thanks for explaining the need for tini, forgot about other container runtimes.

**Comment by nothub** [comment]
> Yes, was talking about the whitelist-vs-blacklist approach here. You bring a good point about ignoring future non-source code files. Your approach is ideal for projects with a unique source code path, usually `src`. Minetrack doesn't (currently) follow that convention, and source code additions are more common than build ones.

Okay, yeah that totally makes sense, i gonna change the .dockerignore file to use a default blacklist approach in this case.

> Is this an established pattern? I'm only used to `.gitignore`s, where the blacklist approach is the standard.

The whitelist approach is not really widely used but for me its the default choice because i think its easier to comprehend and also because this strategy is less prone to usage mistakes (an image without a needed application file will fail on start or will at least fail a test, an unneeded file in an image is easier to miss).

> Re port: most people run Minetrack behind a Nginx proxy on the same machine. However, I doubt this is the case on a K8s setup. I'm okay-ish with it, as long as it's documented in the readme (currently the `-p` flag doesn't clear out what the default is).

Yeah that makes sense. I gonna add a small explanation for the parameters and a second docker run command line with the alternative port.

**Comment by nothub** [comment]
> Re port: most people run Minetrack behind a Nginx proxy on the same machine.

Speaking of, should i also add a small nginx example config snippet? I gonna add it with the readme change and if it is unwanted i can remove it again.

**Review by hugmanrique — APPROVED** [review]

**Comment by nothub** [comment]
d3245e1b268fcf61dfa1fc20c7a71d8f1181a1d6 fixes a problem i have introduced with the rootless user.
With this change, log and database file are writeable again :see_no_evil:

**Comment by Cryptkeeper** [comment]
I do not understand much about Docker, but I trust @hugmanrique 😛

Thanks for the contribution @nothub

**Pull request merged**